eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -50,12 +71,37 @@ foreach ($branch in $branches)
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
-  }
-  catch {
-    LogError "Remove-GitHubSourceReferences failed with exception:`n$_"
-    exit 1
+  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request."
+  # Allow to override the the var $LastCommitOlderThan
+  if ($LastCommitOlderThan -or $env:LASTCOMMITDATEOVERRIDE) {
+    $LastCommitOlderThan = If (!$LastCommitOlderThan) { $env:LASTCOMMITDATEOVERRIDE } Else { $LastCommitOlderThan } 
+    LogDebug "Deleteing the branch which last commit older than the date $LastCommitOlderThan."
+    
+    if (!$res.object -or !$res.object.url) {
+      LogDebug "No commit url returned from response. Skipping... "
+      continue
+    }
+    $GetCommitSHAUrl = $res.object.url
+    $responseJson = HandleGHAPICall -api $GetCommitSHAUrl
+    if ($null -eq $responseJson) {
+      LogError "Error calling the commit SHA API. Exiting..."
+      exit 1
+    }
+    
+    $commitResponse = ConvertFrom-Json $responseJson
+    if (!$commitResponse.author -or !$commitResponse.author.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    if ($commitResponse.author.date -gt $LastCommitOlderThan) {
```

Comment:
This should treat both statements as `[datetime]` objects: 

Running this command in the CLI I get different results depending on how I cast the item on the left: 

```
C:\Users\djurek> '2022-04-23T04:47:03Z' -gt [DateTime]::now    # This doesn't make sense, 4/23 was Saturday
True
C:\Users\djurek> '2022-04-23T04:47:03Z' -lt [DateTime]::now     # Still doesn't make sense
False
C:\Users\djurek> [DateTime]'2022-04-23T04:47:03Z' -lt [DateTime]::now  # OK now it makes sense that 4/23 is less than the current time.
True
C:\Users\djurek>
```

```suggestion
    if ([datetime]$commitResponse.author.date -gt [datetime]$LastCommitOlderThan) {
```

The cast on the right is only necessary if `$LastCommitOlderThan` is not a `[datetime]`


I believe this is happening because PowerShell casts the item on the right side of the expression to the type of the object on the left side of the expression. Flipping the condition could also fix this but I prefer being explicit here.